src/cpp/core/r_util/RSourceIndex.cpp

```diff
@@ -366,6 +366,42 @@ void libraryCallIndexer(const RTokenCursor& cursor,
    }
 }
 
+void testThatCallIndexer(const RTokenCursor& cursor,
+                         const IndexStatus& status,
+                         bool isReadOnlyFile, 
+                         RSourceIndex* pIndex)
+{
+   if (!cursor.isType(RToken::ID))
+      return;
+   
+   if (!(cursor.contentEquals(L"test_that")))
+      return;
+   
+   RTokenCursor clone = cursor.clone();
+   if (!clone.moveToNextToken())
+      return;
+   
+   if (!clone.isType(RToken::LPAREN))
+      return;
+   
+   if (!clone.moveToNextToken())
+      return;
+   
+   if (clone.isType(RToken::STRING))
+   {
+      RSourceItem item(
+         RSourceItem::Test,
+         boost::join(std::vector<std::string> {"test_that(", clone.contentAsUtf8(), ")"}, ""),
```

Comment:
Maybe it's not necessary to prefix with "test_that" but it gives a nice way to fuzzy search for tests, e.g. one can start with "t_"

Reply:
(see other comments; I wonder if we should consider a way of extending the fuzzy finder to allow simple filters for search results, e.g. `t test` to search only tests)